codegen/src/main/kotlin/norm/SqlAnalyzer.kt

```diff
@@ -37,15 +38,22 @@ class SqlAnalyzer(private val connection: Connection) {
     fun sqlModel(namedParamSql: String): SqlModel {
 
         val paramNames = namedParamsRegex.findAll(namedParamSql).map { it.value }.toList()
+        val paramsIndexMapping = paramNames.foldIndexed(HashMap<String, List<Int>>(), { i, acc, param ->
+            acc[param] = acc[param]?.let { it + (i + 1) } ?: listOf(i + 1)
```

Comment:
1. Seems like mutation. try to avoid updating.
2. Extract logic as method so that it is more readable.